filament/src/ShadowMap.cpp

```diff
@@ -1197,11 +1205,34 @@ backend::Viewport ShadowMap::getViewport() const noexcept {
     // We set a viewport with a 1-texel border for when we index outside the
     // texture. This can only happen for the directional light when "focus shadow casters is used".
     const uint32_t dim = mOptions->mapSize;
-    if (isDirectionalShadow()) {
-        return { 1, 1, dim - 2, dim - 2 };
-    } else {
-        return { 0, 0, dim, dim };
+    const uint16_t border = 1u;
+    return { border, border, dim - 2u * border, dim - 2u * border };
```

Comment:
Could we merge `getViewport` and `getScissor` into a single function? Do we really need a separate `getViewport` function?